coordinator-common/src/main/java/org/apache/kafka/coordinator/common/runtime/CoordinatorRuntime.java

```diff
@@ -666,27 +668,25 @@ private void transitionTo(
                     break;
 
                 case ACTIVE:
-                    state = CoordinatorState.ACTIVE;
                     highWatermarklistener = new HighWatermarkListener();
                     partitionWriter.registerListener(tp, highWatermarklistener);
                     coordinator.onLoaded(metadataImage);
                     break;
 
                 case FAILED:
-                    state = CoordinatorState.FAILED;
                     unload();
                     break;
 
                 case CLOSED:
-                    state = CoordinatorState.CLOSED;
                     unload();
                     break;
 
                 default:
+                    // Revert the state update
+                    state = oldState;
+                    runtimeMetrics.recordPartitionStateChange(newState, oldState);
```

Comment:
I think that we could change this switch to using the new switch expression now. A switch expression requires to handle the cases and we handle all the states we could remove the default.